src/pages/DirectMessage.tsx

```diff
@@ -201,35 +246,52 @@ const DirectMessage = () => {
           </Box>
           <Divider />
           <Box sx={styles.chatContainer}>
-            <List component="div" ref={messageRef} sx={styles.messageList}>
-              {messages.map((o) => (
-                <ListItem component="div" key={o.messageId}>
-                  <ListItemText
-                    sx={
-                      o.sentUserId === user.userId
-                        ? styles.sentMessage
-                        : styles.message
-                    }
-                    disableTypography
-                    primary={
-                      <Box
-                        sx={
-                          o.sentUserId === user.userId
-                            ? styles.sentMessageText
-                            : styles.messageText
-                        }
-                      >
-                        <Typography>{o.textContent}</Typography>
-                      </Box>
-                    }
-                    secondary={
-                      <Typography sx={styles.timestamp} variant="body2">
-                        {formatTimestamp(o.timestamp)}
-                      </Typography>
-                    }
-                  />
-                </ListItem>
-              ))}
+            <List
+              component="div"
+              ref={messageRef}
+              sx={styles.messageList}
+              id={"scrollable"}
+            >
+              <InfiniteScroll
+                dataLength={messages.length}
+                next={fetchNextPage}
+                style={{ display: "flex", flexDirection: "column-reverse" }}
+                inverse={true}
+                hasMore={hasNextPage}
+                loader={<PageLoader />}
+                scrollableTarget={"scrollable"}
```

Comment:
Same small nitpick here, we don't need the curly braces for string props
```suggestion
                scrollableTarget="scrollable"
```

Reply:
Done